### ISSUE TITLE
Multiple containers

### ISSUE THREAD

**Issue by reinos** [nuxt-community/gtm-module#10]
Hi there,

Is it possible to have multiple containers. Currently it seems to have only one...

Let me know and when this will be implemented.

Best,
Rein

**Comment by reinos** [comment]
Just opened an PR that will fix this issue (https://github.com/nuxt-community/gtm-module/pull/11)

**Comment by pi0** [comment]
Hi. It should be now fully possible by 2.2.0

**Comment by reinos** [comment]
Thanks @pi0 

Gonna test it today and will report back to you 😄

**Comment by reinos** [comment]
Seems not to working. And by reading the code it will never happend.

```
  if(!window._gtm_init){window._gtm_init=1;
```

The code above is used to check if the GTM has been init. But after the first time `window._gtm_init` is alwasy 1.

This PR https://github.com/nuxt-community/gtm-module/pull/11 seems still the valid option to fix this issue.

Please have a look and consider this solution.

**Comment by pi0** [comment]
@reinos `_gtm_init` is a flag to load utils not container. Also having two `<script>` with same `gtm-script` is so strange! Any chance you can share your project (or a simple repro) with me?

**Comment by reinos** [comment]
Better question; how would I add multiple containers to the page?

Like this?

```
{
  buildModules: [
    '@nuxtjs/gtm',
  ],
  gtm: [
    {
      id: 'GTM-XXXXXXX'
    }, 
    {
      id: 'GTM-XXXXXXX'
    }
  ]
}
```

I notice that I had the configuration inside the `modules` section (we are upgrading from an older nuxtJS...)

**Comment by pi0** [comment]
@reinos For additional containers, you can use a plugin which calls `$gtm.init(<id>)`

`plugins/gtm.js`:

```js
export default function({ $gtm }) {
 $gtm.use('gtm-XXX')
}
```

**Comment by reinos** [comment]
Ok, I have this

config:
```
{
  buildModules: [
    '@nuxtjs/gtm',
  ],
  gtm: {
    layer: 'dataLayer',
    pageTracking: true,
    dev: process.env.NODE_ENV !== 'production',
    autoInit: false,
  }
}
```

Added an plugin `plugins/gtm.js` with 

```
export default function({ $gtm }) {
 $gtm.init('gtm-my-id')
 $gtm.init('gtm-my-id')
}
```

then I get this error `Cannot read property 'find' of undefined` for 

```
const gtmIframe = ctx.app.head.noscript.find(s => s.hid = 'gtm-noscript')
```

**Comment by reinos** [comment]
And how can we use the `push` event for a specific container? https://github.com/nuxt-community/gtm-module#pushing-events

**Comment by pi0** [comment]
Please refer to google [docs](https://developers.google.com/tag-manager/devguide#multiple-containers):

> Note that you can use only a single common data layer for all Google Tag Manager containers on a page because using more than one data layer can cause some triggers to stop working and could have other implications

The cannot read property error seems a little bit odd. What Nuxt version are you using?

**Comment by reinos** [comment]
We are using version `2.11.0`

But try it for yourself, just a have a default config where you omit the `ID` and use a `plugin` to load the 2 IDs should show the error. 

But when you have a fake ID in the config It will work.

```
{
  buildModules: [
    '@nuxtjs/gtm',
  ],
  gtm: {
    layer: 'dataLayer',
    pageTracking: true,
    dev: process.env.NODE_ENV !== 'production',
    autoInit: false,
    id: 'GTM-xxx'
  }
}
```

The only downside is that you have an extra `noscript` iframe and you have to set it in your config.

Also we know that Google discourage the use of 2 containers on the same page, but it is allowed 😉

**Comment by pi0** [comment]
@reinos There was a mistake in module code causing this error fixed by [v2.2.2](https://github.com/nuxt-community/gtm-module/releases/tag/v2.2.2)

Multiple containers allowed but should push to single common data layer :)

**Comment by reinos** [comment]
> There was a mistake in module code causing this error fixed by v2.2.2

Great that fixed the issue

> Multiple containers allowed but should push to single common data layer :)

Yes and how can we manage that in NuxtJS? Can we target a specific container like this? `this.$gtm.get('GTM-xxxx').push(...)`

**Comment by pi0** [comment]
> Yes and how can we manage that in NuxtJS? Can we target a specific container like this? this.$gtm.get('GTM-xxxx').push(...)

I'm not sure if that would be possible by GTM (with/without Nuxt) as we only have one dataLayer.

**Comment by reinos** [comment]
Ok, its a test for our marketing team and the second GTM container will be removed next month. So no big deal.

Case closed, thanks for your support @pi0 !

**Comment by reinos** [comment]
hi @pi0 

Coming back to this issue as it seems to have an overload of tag calls..

Any idea what is going on?

It seems the plugin is loaded for every component or so?

<img width="1626" alt="Screenshot 2020-03-03 at 15 16 44" src="https://user-images.githubusercontent.com/633730/75787076-79d0dd00-5d66-11ea-967a-c8e4f07f1154.png">

**Comment by pi0** [comment]
@reinos Please use `nuxt-edge` for now. Nuxt bug is fixed with 2.12 releasing ASAP :)

**Comment by drewbaker** [comment]
```
export default function({ $gtm }) {
 $gtm.init('gtm-XXX')
 $gtm.init('gtm-YYY')
}
```

Would be good to have that added to the README @pi0
